src/cpu/flexcpu/simple_dataflow_cpu.hh

```diff
@@ -0,0 +1,242 @@
+     * of a given instruction within a given context. Upon completion of the
+     * execution of the instruction, this CPU implementation should notify the
+     * requester by calling the callback function and passing any fault state
+     * resulting from execution.
+     * 
+     * @param inst Reference to an instruction to execute.
+     * @param context Reference to a context interface in which to execute the
+     *  instruction
+     * @param callback_func The function reference for purposes of handling the
+     *  result of the execution.
+     * 
+     * @return Whether or not this request has been acknowledged. If this
+     *  function returns false, the requester should not expect callback_func
+     *  to be called in the future. May return false as a result of 
+     */
+    bool requestExecution(StaticInst& inst, ExecContext& context,
+        std::function<void(Fault)> callback_func);
+// TODO decide if memory access should happen through this function, or a
+// separate dedicated requestDataAddrTranslation/requestData function pair.
+// I'm leaning toward keeping it in this function, since there isn't a logical
+// difference between beginning and finishing execution as far as the status
+// a.k.a. committability of the function goes, if the callback simply doesn't
+// occur until after completeAcc().
+
+    /**
+     * Event-driven means for other classes to request a translation of a
+     * particular address via this CPU's ITB. Upon completion of this
+     * translation (immediate or not), this CPU implementation should return
+     * the values retrieved from the ITB via the callback function, and whether
+     * or not there was any fault as a result of the request.
+     * 
+     * @param inst_vaddr The virtual address we want translated
+     * @param callback_func The function reference for purposes of handling the
+     *  result of the translation
+     * 
+     * @return Whether or not this request has been acknowledged. If this
+     *  function returns false, the requester should not expect callback_func
+     *  to be called in the future.
+     */
+    bool requestInstAddrTranslation(Addr inst_vaddr,
+        std::function<void(Fault,BaseTLB::Translation&)> callback_func);
+
+    /**
+     * Event-driven means for other classes to request a single MachInst via
+     * this CPU's instruction Port.
+     * 
+     * @param inst_paddr A physical address we can create a Request with to
+     *  send through the Port
+     * @param callback_func The function reference for purposes of handling the
+     *  result of the memory request
+     * 
+     * @return Whether or not this request has been acknowledged. If this
+     *  function returns false, the requester should not expect callback_func
+     *  to be called in the future.
+     */
+    bool requestInstruction(Addr inst_paddr,
+        std::function<void(Fault,TheISA::MachInst)> callback_func);
```

Comment:
Again, I'm very excited about this callback function idea. However, my gut says there's a chance that it won't work out.